mlir/include/mlir/Linker/LLVMLinkerMixin.h

```diff
@@ -340,29 +340,38 @@ class LLVMLinkerMixin {
       return ConflictResolution::LinkFromSrc;
     }
 
-    std::optional<ComdatSelector> srcComdatSel = derived.getComdatSelector(pair.src);
-    std::optional<ComdatSelector> dstComdatSel = derived.getComdatSelector(pair.dst);
+    std::optional<ComdatSelector> srcComdatSel =
+        derived.getComdatSelector(pair.src);
+    std::optional<ComdatSelector> dstComdatSel =
+        derived.getComdatSelector(pair.dst);
     if (srcComdatSel.has_value() && dstComdatSel.has_value()) {
       auto srcComdatName = srcComdatSel->name;
       auto dstComdatName = dstComdatSel->name;
       auto srcComdat = srcComdatSel->kind;
       auto dstComdat = dstComdatSel->kind;
       if (srcComdatName != dstComdatName) {
-          llvm_unreachable("Comdat selector names don't match");
+        llvm_unreachable("Comdat selector names don't match");
       }
       if (srcComdat != dstComdat) {
-          llvm_unreachable("Comdat selector kinds don't match");
+        llvm_unreachable("Comdat selector kinds don't match");
       }
 
       if (srcComdat == mlir::LLVM::comdat::Comdat::Any) {
-          return ConflictResolution::LinkFromDst;
+        return ConflictResolution::LinkFromDst;
       }
       if (srcComdat == mlir::LLVM::comdat::Comdat::NoDeduplicate) {
-          return ConflictResolution::Failure;
+        return ConflictResolution::Failure;
       }
       llvm_unreachable("unimplemented comdat kind");
     }
 
+    // If we reach here, we have two external definitions that can't be resolved
+    // This is typically an error case in LLVM linking
+    if (isExternalLinkage(srcLinkage) && isExternalLinkage(dstLinkage) &&
+        !srcIsDeclaration && !dstIsDeclaration) {
+      return ConflictResolution::Failure;
+    }
+
```

Comment:
I'm assuming this would fail the link entirely (`ConflictResolution::Failure`). Correct me if I'm wrong.

Shouldn't we do this below?
```suggestion
    // In the case of two conflicted external definitions or declarations, we stick
    // to the symbol that was already successfully compiled
    if (isExternalLinkage(srcLinkage) && isExternalLinkage(dstLinkage)) {
      return ConflictResolution::LinkFromDst;
    }
```
In the case of `analyze-image` this would also fix "duplicate external symbols", no? If you look in the compilation, you'll see some libraries being included twice. Would that be considered as us having "duplicate external symbols"? I think this change should also preserves the symbol that was already successfully linked.

I tested out the suggested change on my machine and it worked fine for `analyze-image`. I was successful able to compile it, along with passing the `poller` tests. 

I think this should also follow the "first-win" policy linker's use for resolving duplicated symbols.